src/rbd.cc

```diff
@@ -2550,6 +2553,198 @@ static int parse_map_options(char *options)
   return 0;
 }
 
+static int disk_usage_callback(uint64_t offset, size_t len, int exists,
+                               void *arg) {
+  uint64_t *used_size = reinterpret_cast<uint64_t *>(arg);
+  if (exists) {
+    (*used_size) += len;
+  }
+  return 0;
+}
+
+static int compute_image_disk_usage(const std::string& name,
+                                    const std::string& snap_name,
+                                    const std::string& from_snap_name,
+                                    librbd::Image &image, uint64_t size,
+                                    TextTable& tbl, Formatter *f,
+                                    uint64_t *used_size) {
+  const char* from = NULL;
+  if (!from_snap_name.empty()) {
+    from = from_snap_name.c_str();
+  }
+
+  uint64_t flags;
+  int r = image.get_flags(&flags);
+  if (r < 0) {
```

Comment:
@jdurgin librbd's get_flags() just pulls the current state of the flags from ImageCtx.  The only time the ```get_flags``` OSD method is invoked is during an image refresh and that is protected w/ EOPNOTSUPP/EIO checks.